### PR TITLE
[Enhancement] remove be debuginfo file from the final be/cn/allin1 docker image

### LINKED ISSUE #30334: [DockerImage] reduce be/cn/allin1 docker image by removing the starrocks_be.debuginfo
## Enhancement

`starrocks_be.debuginfo` contains the debug symbol that is necessary for be crash analyzing by gdb tool.

However, in container environment, it is hard to preserve the core, not mentioned attaching gdb to do trouble shooting, the `starrocks_be.debuginfo` is useless, just a waste of time pulling images and a waste disk space.

Users can still get the `starrocks_be.debuginfo` from the corresponding artifacts docker image.

For example, to get the debuginfo file for docker image `starrocks/be-ubuntu:3.1.2`, user can run following command to get the corresponding debuginfo file

`docker run --rm starrocks/artifacts-ubuntu:3.1.2 cat /release/be_artifacts/be/lib/starrocks_be.debuginfo > starrocks_be.debuginfo`

* replace `ubuntu` with `centos7` if running with `*-ubi` image
* replace `3.1.2` with the corresponding release version

### PR DESCRIPTION
**Pull request by kevincai**
Fixes #30334

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [X] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [X] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [X] I have checked the version labels which the pr will be auto-backported to the target branch
  - [X] 3.1
  - [X] 3.0
  - [X] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Review by choury — APPROVED** [review]

**Review by wangruin — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

File: docker/dockerfiles/allin1/allin1-ubuntu.Dockerfile [hunk 1]
```diff
@@ -24,6 +24,7 @@ COPY ${LOCAL_REPO_PATH}/fs_brokers/apache_hdfs_broker/output/apache_hdfs_broker
 
 
 FROM artifacts-from-${ARTIFACT_SOURCE} as artifacts
+RUN rm -f /release/be_artifacts/be/lib/starrocks_be.debuginfo
```
Comment (dengliu) [inline docker/dockerfiles/allin1/allin1-ubuntu.Dockerfile:27]: @kevincai  actually it's good to keep it in the allin1. 
in a case we have crash on BE deployed in k8s, we could download the coredump locally and run gdb with the debug info file in allin1 container

Comment (kevincai) [reply docker/dockerfiles/allin1/allin1-ubuntu.Dockerfile:27]: it's alright, you can still pull the debuginfo from artifacts image. :)

**Pull request merged**